contracts/InterchainTokenFactory.sol

```diff
@@ -40,7 +39,7 @@ contract InterchainTokenFactory is IInterchainTokenFactory, ITokenManagerType, M
         interchainTokenService = IInterchainTokenService(interchainTokenService_);
 
         chainNameHash = interchainTokenService.chainNameHash();
-        gateway = interchainTokenService.gateway();
+        gateway = interchainTokenService.getGatewayWithToken();
```

Comment:
```suggestion
        gateway = IAxelarGMPGatewayWithToken(interchainTokenService.gateway());
```

Reply:
this is going to be removed anyways since gateway registration needs to be dropped